### PR TITLE
 Rename ShouldKeepListStructure

### PR DESCRIPTION
**Pull request by ramramps**
### Purpose
 https://jira.autodesk.com/browse/QNTM-697

* Renamed ShouldKeepListStructure to KeepListStructureon PortModel
* Fixed the tests.
* Validated : XML Serialzation / DeSerialization
### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### CODE REVIEW

**Comment by gregmarr** [comment]
I'm not sure why this Jira task says that this is to match the CoGS structure:

https://git.autodesk.com/AEC-Generative-Design/ComputationGraphService/blob/master/api/service.yaml#L1778

```
      ShouldKeepListStructure:
        type: "boolean"
        description: "Whether data from this node will be re-aligned into the original structure of the nested list."
```

**Comment by gregmarr** [comment]
In #7747, @ikeough is proposing the rename in both Dynamo and CoGS.  I'm not sure of the purpose of this rename.

**Comment by ramramps** [comment]
@gregmarr There are other renames in the current sprint. May be you should take a look at them.

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@
 // to distinguish one build from another. AssemblyFileVersion is specified
 // in AssemblyVersionInfo.cs so that it can be easily incremented by the
 // automated build process.
-[assembly: AssemblyVersion("2.0.0.3911")]
+[assembly: AssemblyVersion("2.0.0.5176")]
```
Comment (mjkkirschner) [inline src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs]: don't commit this file - it just causes merge conflicts.

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs [hunk 2]
```diff
@@ -45,7 +45,9 @@
 // to distinguish one build from another. AssemblyFileVersion is specified
 // in AssemblyVersionInfo.cs so that it can be easily incremented by the
 // automated build process.
+
 [assembly: AssemblyVersion("2.0.0.4057")]
+
```
Comment (ramramps) [inline src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs]: removed the merge conflicts.

File: tools/NuGet/BuildPackages.bat [hunk 3]
```diff
@@ -1,23 +1,14 @@
-﻿:: Argument %1: path to template folder
```
Comment (QilongTang) [inline tools/NuGet/BuildPackages.bat]: Looks like you rolled this file into a previous version, can you revert the change to this file only?

Comment (mjkkirschner) [inline tools/NuGet/BuildPackages.bat]: yep, can you try to get rid of this change.

File: tools/NuGet/BuildPackages.bat [hunk 4]
```diff
@@ -54,4 +54,4 @@ for %%f in (nuspec\*.nuspec) do (
   if not exist %%~nf.%version%.nupkg (
     exit /b 1
   )
-)
+)
```
Comment (ramramps) [inline tools/NuGet/BuildPackages.bat]: got rid of that change...this file is same as master.

**Comment by ramramps** [comment]
Merging this now.
